packages/wrangler/src/dev/start-server.ts

```diff
@@ -131,26 +143,75 @@ export async function startDevServer(
 
 				return { accountId, apiToken: requireApiToken() };
 			},
+			persist: props.localPersistencePath ?? undefined,
+			testScheduled: props.testScheduled,
+			registry: workerDefinitions,
 		},
 		build: {
-			// format: props.entry.format,
+			bundle: !props.noBundle,
+			define: props.define,
+			jsxFactory: props.jsxFactory,
+			jsxFragment: props.jsxFragment,
+			tsconfig: props.tsconfig,
+			minify: props.minify,
+			processEntrypoint: props.processEntrypoint,
+			additionalModules: props.additionalModules,
 			moduleRoot: props.entry.moduleRoot,
-			nodejsCompatMode: null,
+			moduleRules: props.rules,
+			nodejsCompatMode: props.nodejsCompatMode,
 		},
 	};
 
-	// temp: fake these events by calling the handler directly
-	if (!props.experimentalDevEnv) {
-		devEnv.proxy.onConfigUpdate({
-			type: "configUpdate",
-			config: fakeResolvedInput(startDevWorkerOptions),
-		});
-		devEnv.proxy.onBundleStart({
-			type: "bundleStart",
-			config: fakeResolvedInput(startDevWorkerOptions),
+	if (props.experimentalDevEnv) {
```

Comment:
Made this function clearer by consolidating into one if-block and returning early when props.experimentalDevEnv is set. The rest of the function is for when it is not set